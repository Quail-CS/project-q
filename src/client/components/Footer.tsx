/**
 * @module Footer.tsx
 * @description Application Footer Presentation Component
 */

import * as React from 'react';
import { Link } from 'react-router-dom';
import Disclaimer from './Disclaimer';

const Footer = () => {
  return (
    <div className="footer">
      <ul className="footer-links">
        <li key="privacy-policy"><Link to='/terms/privacy'>PRIVACY POLICY</Link></li>
        <li key="terms-of-service"><Link to='/terms/service'>TERMS AND CONDITIONS</Link></li>
      </ul>
      <Disclaimer />
    </div>
  );
};

export default Footer;
