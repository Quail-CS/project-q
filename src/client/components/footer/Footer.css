/* Footer - 4vh + 4vh = 8vh */
.footer {
  color: #666666;
  display: grid;
  grid-template-rows: 1fr 1fr;
}

/* Footer Links */
.footer-links {
  letter-spacing: 1.5px;
  list-style-type: none;
  margin: 0;
  padding: 0;

  display: flex;
  flex-flow: row wrap;
  justify-content: center;
  align-items: center;
}

.footer-links li a {
  color: inherit;
  font-size: 0.8rem;
  padding: 4px 20px;
  text-decoration: none;
}

.footer-links a:hover {
  color: #FFFFFF;
}

/* Disclaimer */
.disclaimer {
  display: flex;
  align-items: center;
  justify-content: center;
}

.disclaimer-text {
  font-size: 0.4rem;
  padding: 0 5vw;
  text-align: center;
}