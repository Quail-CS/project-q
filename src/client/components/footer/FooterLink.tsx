/**
 * @module Footer.tsx
 * @description Application Footer Link Presentation Component
 * UNIT TEST COVERAGE - 100%
 */

import * as React from 'react';
import { Link } from 'react-router-dom';

export interface FooterLinkProps {
  linkKey: string,
  link: string,
  text: string,
}

const FooterLink = (props: FooterLinkProps): JSX.Element => {
  const { linkKey, link, text } = props;
  return (
    <li key={linkKey}><Link to={link}>{text}</Link></li>
  )
}

export default FooterLink;
