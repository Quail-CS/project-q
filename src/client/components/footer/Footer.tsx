/**
 * @module Footer.tsx
 * @description Application Footer Presentation Component
 * UNIT TEST COVERAGE - 100%
 */

import * as React from 'react';
import Disclaimer from './Disclaimer';
import FooterLink from './FooterLink';

import './Footer.css';

// TODO: REFACTOR
const footerLinkArray: JSX.Element[] = [];
export const footerLinkData: string[][] = [
  ['privacy-policy', '/terms/privacy', 'PRIVACY POLICY'],
  ['terms-of-service', '/terms/service', 'TERMS AND CONDITIONS'],
  ['frequently-asked-questions', '/terms/faq', 'FAQ'],
];
footerLinkData.forEach(data => footerLinkArray.push(<FooterLink key={data[0]} linkKey={data[0]} link={data[1]} text={data[2]} />))

const Footer = () => {
  return (
    <div className="footer">
      <ul className="footer-links">
        {footerLinkArray}
      </ul>
      <Disclaimer />
    </div>
  );
};

export default Footer;
