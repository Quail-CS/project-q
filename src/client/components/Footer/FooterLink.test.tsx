import * as React from 'react';
import * as Enzyme from 'enzyme';
import * as Adapter from 'enzyme-adapter-react-16';
import * as Renderer from 'react-test-renderer';

import FooterLink, { FooterLinkProps } from './FooterLink';
import { MemoryRouter } from 'react-router';
import { footerLinkData } from './Footer';
import { Link } from 'react-router-dom';

Enzyme.configure({ adapter: new Adapter() });

const props: FooterLinkProps = {
  linkKey: footerLinkData[0][0],
  link: footerLinkData[0][1],
  text: footerLinkData[0][2],
}

describe('FooterLink Component', () => {
  it('matches the snapshot', () => {
    const tree: Renderer.ReactTestRendererJSON = Renderer.create(
      <MemoryRouter>
        <FooterLink {...props} />
      </MemoryRouter>
    ).toJSON() as Renderer.ReactTestRendererJSON;
    expect(tree).toMatchSnapshot();
  })
  it('renders a <li> with key equal to linkKey prop', () => {
    const footerLinkWrapper: Enzyme.ShallowWrapper = Enzyme.shallow(<FooterLink {...props} />)
    expect(footerLinkWrapper.type()).toEqual('li');
    expect(footerLinkWrapper.key()).toEqual(props.linkKey);
  })
  it('contains a Link to the link prop with text equal to the text prop', () => {
    const footerLinkWrapper: Enzyme.ShallowWrapper = Enzyme.shallow(<FooterLink {...props} />)
    expect(footerLinkWrapper.find(Link)).toHaveLength(1);
    expect(footerLinkWrapper.find(Link).props()).toHaveProperty('children', props.text)
    expect(footerLinkWrapper.find(Link).props()).toHaveProperty('to', props.link)
  })
});
