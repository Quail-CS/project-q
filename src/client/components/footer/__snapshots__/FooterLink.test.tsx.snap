// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FooterLink Component matches the snapshot 1`] = `
<li>
  <a
    href="/terms/faq"
    onClick={[Function]}
  >
    FAQ
  </a>
</li>
`;
